src/java.desktop/share/classes/javax/swing/ImageIcon.java

```diff
@@ -296,6 +296,12 @@ protected void loadImage(Image image) {
         synchronized(mTracker) {
             int id = getNextID();
 
+            if (id < 0) {
+                loadStatus = MediaTracker.ABORTED;
+                width = image.getWidth(imageObserver);
+                height = image.getHeight(imageObserver);
+                return;
+            }
             mTracker.addImage(image, id);
```

Comment:
Hmm. Is there any acceptable logging we could use here?

(I'm just thinking: from the perspective of a developer trying to debug a customer complaint, this would be a lot easier to identify in System.err mentioned it...? Otherwise I'd start by trying to look for potential memory leaks or other red herrings.)

Reply:
I dont think there is any logging for this nor do we want to log...In all probability, the image will consume the heap memory so practically OOM will be raised by the time this overflow happens but it can happen if maybe all 1x1 images are used and subsequent needed memory reserved for Java process

Reply:
> OK. This feels too theoretical for me to invest strongly in a recommendation.

That's why I mentioned in the description itself 

> Theoretically there is a possibility that there can be overflow in the long time run or for large number of created "imageIcon"

Reply:
What's the deal with OOM at all?

If the application doesn't keep all the images it loads, no OOM will occur. Throwing OOM has nothing to do with the stated problem.